modules/delegates.js

```diff
@@ -687,17 +652,19 @@ Delegates.prototype.updateForgingStatus = function(
  * Gets delegate list based on input function by vote and changes order.
  *
  * @param {number} round
- * @param {function} source - Source function for get delegates
  * @param {function} cb - Callback function
  * @param {Object} tx - Database transaction/task object
  * @returns {setImmediateCallback} cb, err, truncated delegate list
  * @todo Add description for the params
  */
-Delegates.prototype.generateDelegateList = function(round, source, cb, tx) {
+Delegates.prototype.getForgersList = function(round, cb, tx) {
 	// Set default function for getting delegates
-	source = source || __private.getKeysSortByVote;
-
-	source((err, truncDelegateList) => {
+	const lastBlockRound = slots.calcRound(modules.blocks.lastBlock.get().height);
+	const source =
```

Comment:
So it's not possible to make this function to decide which delegates list its need to return based only on `round` parameter. Because when we call it from inside of `receiveForkFive` - we need to validate against previous round when `lastBlock.height % ACTIVE_DELEGATES === 0` is `true`, but when we call it from a different source we need to return list for current round always. However, if we pass `height` instead of `round` as a parameter we can do that. 🤗